packages/wonder-blocks-announcer/src/Announcer.ts

```diff
@@ -0,0 +1,257 @@
+// TODO: publish wonder-blocks-style package WB-1776
+// import {srOnly} from "../../wonder-blocks-style/src/styles/a11y";
+
+import {
+    PolitenessLevel,
+    RegionFactory,
+    RegionDictionary,
+    RegionDef,
+} from "../types/Announcer.types";
+
+import {
+    createRegionWrapper,
+    createDuplicateRegions,
+    removeMessage,
+} from "./util/dom";
+import {alternateIndex, debounce} from "./util/util";
+
+const REMOVAL_TIMEOUT_DELAY = 5000;
+const DEFAULT_WAIT_THRESHOLD = 250;
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
+    private static _instance: Announcer | null;
+    node: HTMLElement | null = null;
+    regionFactory: RegionFactory = {
+        count: 2,
+        aIndex: 0,
+        pIndex: 0,
+    };
+    dictionary: RegionDictionary = new Map();
+    stateLock = false;
+
+    private constructor() {
+        if (typeof document !== "undefined") {
+            const topLevelId = `wbAnnounce`;
+            // Check if our top level element already exists
+            const announcerCheck = document.getElementById(topLevelId);
+
+            // Init new structure if the coast is clear
+            if (announcerCheck === null) {
+                this.init(topLevelId);
+            }
+            // The structure exists but references are lost, so help HMR recover
+            else {
+                this.reattachNodes();
+            }
+        }
+    }
+    /**
+     * Singleton handler to ensure we only have one Announcer instance
+     * @returns {Announcer}
+     */
+    static getInstance() {
+        if (!Announcer._instance) {
+            Announcer._instance = new Announcer();
+        }
+        return Announcer._instance;
+    }
+    /**
+     * Internal initializer method to create live region elements
+     * Prepends regions to document body
+     * @param {string} id ID of the top level node (wbAnnounce)
+     */
+    init(id: string) {
+        this.node = document.createElement("div");
+        this.node.id = id;
+        this.node.setAttribute("data-testid", id);
+
+        Object.assign(this.node.style, srOnly);
+
+        // For each level, we create at least two live region elements.
+        // This is to work around AT occasionally dropping messages.
+        const aWrapper = createRegionWrapper("assertive");
+        createDuplicateRegions(
+            aWrapper,
+            "assertive",
+            this.regionFactory.count,
+            this.dictionary,
+        );
```

Comment:
**thought:**  One thing that I noticed while playing with Storybook is that if I change the text in the Story and click the "submit" button, sometimes the announcement is not read by Voice Over. I'm not sure where to put this, but saw that sometimes was announced in one of the duplicate regions.

Reply:
Hmm, that is odd. I noticed this if I change the politeness level to Assertive but not for different text. The default Storybook frame also seems to impact reliability of messages, so I tend to test it in a standalone window. I think I'll opt for integration testing to observe these kinds of issues since I've spent so long on it already!